Use of mismatched enum types.

I see lots of warnings of the following kind:
```
/home/bangerth/p/deal.II/1/projects/aspect/source/particle/manager.cc:473:34: warning: bitwise operation between different enumeration types ‘dealii::EvaluationFlags::EvaluationFlags’ and ‘dealii::UpdateFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  473 |       if (evaluation_flags_union & (update_values | update_gradients))
      |           ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bangerth/p/deal.II/1/projects/aspect/source/particle/manager.cc:483:34: warning: bitwise operation between different enumeration types ‘dealii::EvaluationFlags::EvaluationFlags’ and ‘dealii::UpdateFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  483 |       if (evaluation_flags_union & update_values)
      |           ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/home/bangerth/p/deal.II/1/projects/aspect/source/particle/manager.cc:486:34: warning: bitwise operation between different enumeration types ‘dealii::EvaluationFlags::EvaluationFlags’ and ‘dealii::UpdateFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  486 |       if (evaluation_flags_union & update_gradients)
      |           ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/bangerth/p/deal.II/1/projects/aspect/source/particle/manager.cc:492:38: warning: bitwise operation between different enumeration types ‘dealii::EvaluationFlags::EvaluationFlags’ and ‘dealii::UpdateFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  492 |           if (evaluation_flags_union & update_values)
      |               ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/home/bangerth/p/deal.II/1/projects/aspect/source/particle/manager.cc:496:38: warning: bitwise operation between different enumeration types ‘dealii::EvaluationFlags::EvaluationFlags’ and ‘dealii::UpdateFlags’ is deprecated [-Wdeprecated-enum-enum-conversion]
  496 |           if (evaluation_flags_union & update_gradients)
      |               ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
```
We should probably address this before the release.

--- a/source/particle/manager.cc
+++ b/source/particle/manager.cc
@@ -470,7 +470,7 @@ namespace aspect
       for (unsigned int i=0; i<evaluation_flags.size(); ++i)
         evaluation_flags_union |= evaluation_flags[i];
 
-      if (evaluation_flags_union & (update_values | update_gradients))
+      if (evaluation_flags_union & (EvaluationFlags::values | EvaluationFlags::gradients))
         {
           // Reinitialize and evaluate the requested solution values and gradients
           evaluator.reinit(inputs.current_cell,
@@ -480,20 +480,20 @@ namespace aspect
                              evaluation_flags);
         }
 
-      if (evaluation_flags_union & update_values)
+      if (evaluation_flags_union & EvaluationFlags::values)
         inputs.solution.resize(n_particles,small_vector<double,50>(evaluator.n_components(), numbers::signaling_nan<double>()));
 
-      if (evaluation_flags_union & update_gradients)
+      if (evaluation_flags_union & EvaluationFlags::gradients)
         inputs.gradients.resize(n_particles,small_vector<Tensor<1,dim>,50>(evaluator.n_components(), numbers::signaling_nan<Tensor<1,dim>>()));
 
       for (unsigned int i = 0; i<n_particles; ++i)
         {
           // Evaluate the solution, but only if it is requested in the update_flags
-          if (evaluation_flags_union & update_values)
+          if (evaluation_flags_union & EvaluationFlags::values)
             evaluator.get_solution(i, {&inputs.solution[i][0],inputs.solution[i].size()}, evaluation_flags);
 
           // Evaluate the gradients, but only if they are requested in the update_flags
-          if (evaluation_flags_union & update_gradients)
+          if (evaluation_flags_union & EvaluationFlags::gradients)
             evaluator.get_gradients(i, {&inputs.gradients[i][0],inputs.gradients[i].size()}, evaluation_flags);
         }
 

